nucypher/network/nodes.py

```diff
@@ -1331,10 +1331,15 @@ def nickname_icon_details(self):
             address_first6=self.checksum_address[2:8]
         )
 
-    def known_nodes_details(self) -> dict:
+    def known_nodes_details(self, raise_invalid=True) -> dict:
         abridged_nodes = {}
         for checksum_address, node in self.known_nodes._nodes.items():
-            abridged_nodes[checksum_address] = self.node_details(node=node)
+            try:
+                abridged_nodes[checksum_address] = self.node_details(node=node)
+            except self.StampNotSigned:
+                if raise_invalid:
+                    raise
+                self.log.error(f"encountered unsigned stamp for node with checksum: {checksum_address}")
```

Comment:
I'm leaning towards `warn` instead of `error` since we are ignoring and not raising. Thoughts?

Reply:
The main reason I went with error is so we get good context on sentry.
https://sentry.io/organizations/nucypher/issues/2134995939/?project=1480080&query=is%3Aunresolved